scripts/configure.py

```diff
@@ -0,0 +1,142 @@
+#!/usr/bin/env python 
+
+import os
+import shutil
+import tempfile
+import subprocess
+import urllib
+import json
+import sys
+
+
+def do_install(package_name, requirement=False, target='.'):
+    """Install packages locally without external dependency
+
+    Parameters
+    ----------
+    package_name: string
+        package_name or path/to/MAR-INF/requirements.txt
+    target: string
+        the target local path for installation
+    requirement: bool
+        whether installation is using MAR-INF/requirements.txt
+    """
+    # import pip main robustly against different version of pip and python
+    try:
+        from pip._internal import main
+    except Exception:
+        from pip import main
+    # install requirement or MAR-INF/requirements.txt
+    if not requirement:
+        main(['install', package_name, '-t', target])
+    else:
+        main(['install', '-r', package_name, '-t', target])
+
+
+def download_url(url, target, retries=2, base_retry_interval=0.01):
+    """Download url to a file.
+    Parameters
+    ----------
+    url: string
+        url to the file to download
+    target: string
+        the target local path of the downloaded file
+    retries: int
+        the max number of retries allowed for urlretrieve
+    """
+    assert retries >= 1, "Number of retries should be at least 1"
+
+    retry = 0
+    while retry < retries:
+        try:
+            urllib.urlretrieve(url, target)
+            return
+        except Exception as e:
+            retry += 1
+            if retry < retries: 
+                time.sleep(2 ** retries * base_retry_interval)
+            else:
+                raise e
+
+
+def check_existence(filename, path):
+    """Check the existence of filename in path
+
+    Parameters
+    ----------
+    filename: string
+        name of file or dir to verify existence
+    path: string
+        path to verify the existence
+
+    Returns
+    -------
+    bool: Returns True if exist, False otherwise
+    
+    Examples
+    ----------
+    >>> check_existence('mxnet/', lambda_function_path)
+    >>> check_existence(requirements_path, lambda_function_path)
+    """
+    return os.path.exists(os.path.join(path, filename))
+
+
+def configure(model_archive, lambda_function_path, model_bucket=None):
+    """Configure Lambda Function to consume Model Archive
+    
+    Parameters
+    ----------
+    model_archive: string
+        url or path to the local model archive file
+    model_bucket: string
+        optional S3 bucket name to upload local model archive file
+    """
+    mxnet_dependency_path = 'mxnet/'
+    requirements_path = 'MAR-INF/requirements.txt'
+
+    # check weither it is url
+    is_url = "://" in model_archive
```

Comment:
why not use urlib to validate url?

Reply:
Specifically for urllib, "For Python versions earlier than 2.7.9, urllib does not attempt to validate the server certificates of HTTPS URIs. Use at your own risk!"